### PR TITLE
Add ssh shell unit tests

### PR DESCRIPTION
**Pull request by Ch3LL**

### CODE REVIEW

**Review by s0undt3ch — DISMISSED** [review]

**Review by garethgreenaway — DISMISSED** [review]

File: tests/unit/client/ssh/test_shell.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import os
+import subprocess
+
+import salt.client.ssh.shell as shell
+from tests.support.runtests import RUNTIME_VARS
+from tests.support.unit import TestCase
+
+
```
Comment (s0undt3ch) [inline tests/unit/client/ssh/test_shell.py:8]: Decorate this class with the requires system binaries helper, the windows failures are valid

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by s0undt3ch — DISMISSED** [review]

File: tests/pytests/unit/client/ssh/test_shell.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        """
+        shell.gen_key(keys["priv_key"])
+        for fp in keys.keys():
+            assert os.path.exists(keys[fp])
+
+        # verify there is not a passphrase set on key
+        ret = subprocess.check_output(
+            ["ssh-keygen", "-f", keys["priv_key"], "-y"], timeout=30,
+        )
+        assert ret.decode().startswith("ssh-rsa")
```
Comment (s0undt3ch) [inline tests/pytests/unit/client/ssh/test_shell.py:30]:
```suggestion
import os
import subprocess
import types

import pytest
import salt.client.ssh.shell as shell


@pytest.fixture
def keys(tmp_path):
    pub_key = tmp_path / "ssh" / "testkey.pub"
    priv_key = tmp_path / "ssh" / "testkey"
    return types.SimpleNamespace(pub_key=pub_key, priv_key=priv_key)


@pytest.mark.skip_on_windows(reason="Windows does not support salt-ssh")
@pytest.mark.skip_if_binaries_missing("ssh", "ssh-keygen", check_all=True)
def test_ssh_shell_key_gen(self, keys):
    """
    Test ssh key_gen
    """
    shell.gen_key(str(keys.priv_key))
    assert keys.priv_key.exists()
    assert keys.pub_key.exists()
     # verify there is not a passphrase set on key
    ret = subprocess.check_output(
        ["ssh-keygen", "-f", str(keys.priv_key), "-y"], timeout=30,
    )
    assert ret.decode().startswith("ssh-rsa")
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
Cleaner?

**Review by s0undt3ch — DISMISSED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Comment by krionbsd** [comment]
Should we add `__init__.py` into `tests/pytests/unit/client` ?
It seems `tests/unit/client/test_ssh.py` still exists, it's intended?
`tests/pytests/unit/client/ssh/test_ssh.py` misses `salt.utils.thin` and `salt.utils.parsers` imports

**Pull request merged**